### ISSUE TITLE
"PRAGMA table_info" in SQLite is unreliable when write-ahead logging is enabled

### ISSUE THREAD

**Issue by xapienz** [robolectric/robolectric#5108]
### Description
If database is opened with write-ahead logging enabled, results returned by `PRAGMA table_info` are unreliable. Sometimes outdated results can be returned.

Even though statements are executed sequentially, the result of operations depends on how statements are organized into transactions.
Also, the result depends on sdk version specified in robolectric.properties.

The issue does not reproduce on real device, but happens in Robolectric tests.

### Steps to Reproduce
I try to execute the following sequence of statements:
```
CREATE TABLE IF NOT EXISTS 'mytable' ('foo' INTEGER);
PRAGMA table_info('mytable');    -- should return cursor with 1 row
DROP TABLE 'mytable';
PRAGMA table_info('mytable');    -- should return cursor with 0 rows
```
I enabled write-ahead logging for the database and also wrap some of these statements into transactions.

From each pragma result I output the number of rows in cursor, it should equal to number of columns in table 'mytable'.
Expected result under all conditions should be "1 0", because at the first pragma statement the table has 1 column, and at the second pragma statement it doesn't have columns.

However, the actual result is "0 0", when calling all these statements outside of transactions.
If I wrap all statements into transactions, it correctly returns "1 0".
If I wrap last two statements into transactions, it returns "0 1".

### Robolectric & Android Version
Robolectric 4.3, `sdk=21` in robolectric.properties.
If I change sdk version to 28, results are different from 21, but still not correct.

### Link to a source demonstrating the problem:
https://gist.github.com/xapienz/8deda4dc0dabbfede7800c8b1c2b9624

**Comment by xapienz** [comment]
Looks like problem is that Robolectric is using sqlite 3.7.10 (using com.almworks.sqlite4java), and there was a bug fixed in sqlite 3.7.16:
> 2013-03-18 (3.7.16)
> ...
> 27. Bug fix: Make sure the schema is up-to-date prior to running pragmas table_info, index_list, index_info, and foreign_key_list. 
> https://www.sqlite.org/changes.html

So sqlite binary or com.almworks.sqlite4java may need to be updated

**Comment by LitterSun** [comment]
Robolectric latest version should fix the problem
